lib/xo.ts

```diff
@@ -251,6 +251,30 @@ export class Xo {
 		}
 	}
 
+	/**
+	 * Checks if any directly specified files are ignored and throws an error.
+	 * @private
+	 */
+	async checkForIgnoredDirectFiles(globs: string[]): Promise<void> {
+		const {flatOptions} = await resolveXoConfig(this.linterOptions);
+		const combinedIgnores = [
+			...defaultIgnores,
+			...arrify(this.baseXoConfig.ignores),
+			...flatOptions.flatMap(config => config.ignores ? arrify(config.ignores) : []),
+		];
+
+		// Filter non-glob patterns first (synchronous operation)
+		const directFileGlobs = globs.filter(glob =>
+			!glob.includes('*') && !glob.includes('?') && !glob.includes('['));
```

Comment:
Couldn't direct files also have these strings? `[slug]`.js is common with next.js for instance.